promise.js

```diff
@@ -0,0 +1,45 @@
+var temp = require('./lib/temp')
+var validTemplate = require('./lib/validate-template')
+
+function promisify (fn) {
+  return function () {
+    var that = this
+    var args = new Array(arguments.length)
+
+    for (var i = 0; i < arguments.length; i++) {
+      args[i] = arguments[i]
+    }
+    return new Promise(function (resolve, reject) {
+      args.push(function (err, res) {
+        if (err) return reject(err)
+
+        resolve(res)
+      })
+
+      fn.apply(that, args)
+    })
+  }
+}
+
+function template (template) {
+  if (typeof template !== 'string') {
+    throw new TypeError('template is not a string')
+  }
+
+  if (validTemplate(template) !== true) {
```

Comment:
Could you call `validTemplate` for `validateTemplate` and let that function throw the `TypeError`?